v3_usm.go

```diff
@@ -972,6 +972,12 @@ func (sp *UsmSecurityParameters) unmarshal(flags SnmpV3MsgFlags, packet []byte,
 	}
 	// blank msgAuthenticationParameters to prepare for authentication check later
 	if flags&AuthNoPriv > 0 {
+		// In case if the authentication protocol is not configured or set to NoAuth, then the packet cannot
+		// be processed further
+		if sp.AuthenticationProtocol <= NoAuth {
+			msg := "authentication parameters are not configured to parse incoming authenticated message"
+			return 0, fmt.Errorf("error parsing SNMPv3 User Security Model: %s ", msg)
```

Comment:
I understand the desire to split long lines, but this is basically a static string. We should use `errors.New()` here.

Reply:
Thanks for the comments. I have updated the diff. Can you please review the changes

Reply:
@SuperQ The pipelines seems failing when executing "apt-get update". Can you please help on this to resolve the issue